letsencrypt/client/network2.py

```diff
@@ -90,6 +88,9 @@ def _check_response(cls, response, content_type=None):
                         response_ct)
 
                 try:
+                    # TODO: This is insufficient or doesn't work as intended.
```

Comment:
Boulder delivers `messages2.Error` with all necessary fields populated. Since the exception is not caught anywhere, traceback is printed. It doesn't include any of the fields because `messages2.Error.__str__` is not implemented.

So while I can see it as a our (client) "problem" I don't see that "this is a problem with Boulder". What did you mean and which "another comment" did you refer to? :)